internal/controller/storageclient_controller.go

```diff
@@ -309,26 +402,165 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err != nil {
 				return reconcile.Result{}, fmt.Errorf("failed to create remote noobaa: %v", err)
 			}
-		}
-	}
-	if r.storageClient.GetAnnotations()[storageClaimProcessedAnnotationKey] != "true" {
-		if err := r.reconcileBlockStorageClaim(); err != nil {
-			return reconcile.Result{}, err
-		}
+		case "StorageClass":
+			storageClass := &storagev1.StorageClass{}
+			if err := json.Unmarshal(eResource.Data, &storageClass); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal storage configuration response: %v", err)
+			}
+			// TODO: there will be a clash if storageclass is being reconcile by another entity
+			// ex: in internal mode storageclass was created by storagecluster controller
+			// and if we are mutating anything other than params there'll be unending reconcile loop
+			if err := utils.CreateOrReplace(r.ctx, r.Client, storageClass, func() error {
```

Comment:
- There is no name for StorageClass, createOrReplace will fail
- in the CreateOrReplace the provisioner and the reclaimPolicy is missing

Reply:
no, that data should be coming from provider and unmarshalled before mutation.

Reply:
But we are unmarshalling eResource.Data that contains the parameters isn't it? is there a provider side PR too to have these changes?

Reply:
updated to be consistent, always respect separate labels & annotations from proto and everything else from unmarshalled data, since storage & snapshot classes doesn't have a spec we'll update whatever required (namespace) and consume the remaining of whatever provider sends.